src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -279,4 +291,65 @@ public static void waitSeconds(long second) {
       logger.info("cannot sleep");
     }
   }
+
+  public static boolean urlHasString(String targetString) {
+    String strUrl = SeleniumDriver.getDriver().getCurrentUrl();
+    return strUrl.contains(targetString);
+  }
+
+  public static void reloadPage() {
+    SeleniumDriver.getDriver().navigate().refresh();
+  }
+
+  public static void cleanupDirectory(File dir) {
+    for (File file: dir.listFiles()) {
+      System.out.println(file);
+      if (file.isDirectory()) {
+        cleanupDirectory(file);
+      }
+      file.delete();
+    }
+  }
+
+  public static Map generateDraftFromPipelineJson(String pipelineJsonFileName) throws IOException {
+    Reader reader = Files.newBufferedReader(Paths.get(Constants.FIXTURES_DIR + pipelineJsonFileName));
+    Map jsonMap = GSON.fromJson(reader, Map.class);
```

Comment:
can you change `Map` to `HashMap<K,V>`

Reply:
Any particular reason? better to use the abstract at declaration to have more flexibility further down.